tbx/core/blocks.py

```diff
@@ -501,24 +507,24 @@ def clean(self, value):
         struct_value = super().clean(value)
 
         errors = {}
-        page = value.get("page")
-        link_text = value.get("link_text")
+        button_link = value.get("button_link")
+        button_text = value.get("button_text")
 
-        if page and not link_text:
+        if button_link and not button_text:
             error = ErrorList(
-                [ValidationError("You must add link text for the specified page.")]
+                [ValidationError("You must add button text for the button link.")]
             )
-            errors["link_text"] = error
+            errors["button_text"] = error
 
-        if link_text and not page:
+        if button_text and not button_link:
             error = ErrorList(
                 [
                     ValidationError(
-                        "You must specify a page for the link text you have provided."
+                        "You must specify a button link above, for the button text you have provided."
```

Comment:
**Comment** (non-blocking): I'm always uncertain about overzealous insistence that a user delete the button text if they delete the link. Why can't we allow them to leave the text there for later, when they want to use the link again?

On the other hand this makes sure that the stored data more closely matches the things we use it for, and it's the way we seem to build things.